CONTRIBUTING.md

```diff
@@ -28,3 +28,20 @@ make run
 ```sh
 make uninstall
 ```
+
+## Design choices
+
```

Comment:
### Naming conventions
For most of the resources in the OpenStack API, names are not unique. OpenStack relies in UUIDs for differentiating between resources. This won't be the case for Gopherkube, in which we rely on name uniqueness to differentiate between resources. For such purpose, we create mapping between the resource representation in Gopherkube and the resource in OpenStack.

<<<<
Not sure how does that sound, but I think we should clarify

Also, perhaps we can add some paragraph on the difference between adopting and creating?

Reply:
Brilliant, thanks.

WRT adopting vs creating: you are right, this definitely needs to be here. However since we haven't yet defined how this is going to happen in the MVP, I'd defer that to later. WDYT?